### PR TITLE
Add rule_id to env_yml when jinja processing yaml rule files.

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Add rule_id to env_yml when jinja processing yaml rule files.

#### Rationale:

- `rule_id` can be used within YAML rule files.

### CODE REVIEW

**Review by JAORMX — APPROVED** [review]
exactly what I needed! Thanks!

File: ssg/build_yaml.py [hunk 1]
```diff
@@ -1085,14 +1085,17 @@ def __init__(self, id_):
...
-        yaml_contents = open_and_macro_expand(yaml_file, env_yaml)
-        if yaml_contents is None:
-            return None
-
         rule_id, ext = os.path.splitext(os.path.basename(yaml_file))
         if rule_id == "rule" and ext == ".yml":
             rule_id = get_rule_dir_id(yaml_file)
 
+        if env_yaml:
+            env_yaml["rule_id"] = rule_id
```
Comment (cipherboy) [inline ssg/build_yaml.py:1093]: In most places, prior to modification (such as https://github.com/ComplianceAsCode/content/blob/master/ssg/build_ovals.py#L269 or https://github.com/ComplianceAsCode/content/blob/master/ssg/build_remediations.py#L221) we'd duplicate `env_yaml` and call it `local_env_yaml`. This keeps a constant dict from the caller's perspective (right now, you're modifying the caller's copy as well).

I'd perhaps suggest adding a `self.local_env_yaml = None` member and if `env_yaml` is specified, duplicate it into the local one. Places that load a rule can then use a single consistent env_yaml (populated from the `Rule` class) in most places.

My 2c.

Comment (cipherboy) [reply ssg/build_yaml.py:1093]: That said, I am definitely in favor of this! Thanks @ggbecker!

Comment (ggbecker) [reply ssg/build_yaml.py]: what do you think about 0aa0087c01528a34364f28e3a419000772195351 ?

**Review by cipherboy — APPROVED** [review]

File: ssg/build_yaml.py (hunk 1 above)
Comment (cipherboy) [reply ssg/build_yaml.py]: Looks great, thanks!

**Comment by cipherboy** [comment]
Just for my own edification, I see this gating check failed:

>  Gating / Content Test Filtering on Ubuntu Latest (pull_request) Failing after 23s — Content Test Filtering on Ubuntu Latest 

It failed on the stage "Updating PR":

> Run python3 ./ctf/utility_scripts/comment_pr.py --token *** --pr 7140 --comment ctf.md
> 403 Client Error: Forbidden for url: https://api.github.com/repos/ComplianceAsCode/content/issues/7140/comments
> Error: Process completed with exit code 1.

I reran it and it failed again.

Is this something that must pass before merging? Or what's the process here?

**Comment by ggbecker** [comment]
> Just for my own edification, I see this gating check failed:
> 
> > Gating / Content Test Filtering on Ubuntu Latest (pull_request) Failing after 23s — Content Test Filtering on Ubuntu Latest
> 
> It failed on the stage "Updating PR":
> 
> > Run python3 ./ctf/utility_scripts/comment_pr.py --token *** --pr 7140 --comment ctf.md
> > 403 Client Error: Forbidden for url: https://api.github.com/repos/ComplianceAsCode/content/issues/7140/comments
> > Error: Process completed with exit code 1.
> 
> I reran it and it failed again.
> 
> Is this something that must pass before merging? Or what's the process here?



> Just for my own edification, I see this gating check failed:
> 
> > Gating / Content Test Filtering on Ubuntu Latest (pull_request) Failing after 23s — Content Test Filtering on Ubuntu Latest
> 
> It failed on the stage "Updating PR":
> 
> > Run python3 ./ctf/utility_scripts/comment_pr.py --token *** --pr 7140 --comment ctf.md
> > 403 Client Error: Forbidden for url: https://api.github.com/repos/ComplianceAsCode/content/issues/7140/comments
> > Error: Process completed with exit code 1.
> 
> I reran it and it failed again.
> 
> Is this something that must pass before merging? Or what's the process here?

this one is actually broken right now... we can just ignore. The one that which was supposed to be required is `CaC Jenkins build and unit tests`... but there's some problem with the branch protection. For now the results here should be enough.

**Review by yuumasato — APPROVED** [review]
Looking good, thank you all for improvement and reviews.

**Pull request merged**